sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -0,0 +1,735 @@
+                    {
+                        continue;
+                    }
+
+                    if (hasHttpTrigger)
+                    {
+                        newFunction.IsHttpTrigger = true;
+                    }
+
+                    newFunction.RawBindings = bindings!; // won't be null b/c TryGetBindings would've failed and this line wouldn't be reached
+
+                    result.Add(newFunction);
+                }
+
+                return result; 
+            }
+
+            /// <summary>
+            /// Formats an object into a string value for the source-generated file. This can mean adding quotation marks around the string
+            /// representation of the object, or leaving it as is if the object is a string or Enum type.
+            /// </summary>
+            /// <param name="propValue">The property that needs to be formmated into a string.</param>
+            /// <returns></returns>
+            private string FormatObject(object propValue, bool isEnum = false)
+            {
+                if (propValue is null)
+                {
+                    return "null";
+                }
+
+                // catch values that are already strings or Enum parsing
+                // we don't need to surround these cases with quotation marks
+                if (propValue.ToString().Contains("\"") || isEnum)
+                {
+                    return propValue.ToString();
+                }
+
+                return "\"" + propValue.ToString() + "\"";
+            }
+
+            /// <summary>
+            /// Format an array into a string.
+            /// </summary>
+            /// <param name="enumerableValues">An array object to be formatted.</param>
+            /// <returns></returns>
+            private string FormatArray(IEnumerable enumerableValues)
+            {
+                string arrAsString;
+
+                arrAsString = "new List<string> { ";
+
+                foreach (var o in enumerableValues)
+                {
+                    arrAsString += FormatObject(o);
+                    arrAsString += ",";
+                }
+
+                arrAsString = arrAsString.TrimEnd(',', ' ');
+                arrAsString += " }";
+
+                return arrAsString;
+            }
+
+            /// <summary>
+            /// Checks if a candidate method has a Function attribute on it.
+            /// </summary>
+            private bool IsValidMethodAzureFunction(SemanticModel model, MethodDeclarationSyntax method, out string functionName)
+            {
+                functionName = string.Empty;
+                var methodSymbol = model.GetDeclaredSymbol(method);
+
+                if (methodSymbol is null)
+                {
+                    _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.SymbolNotFound, method.Identifier.GetLocation(), nameof(methodSymbol)));
```

Comment:
I'm not sure how to cause this error in Functions specifically - when I looked into `GetDeclaredSymbol` returning `null` there were a few different errors I saw ppl run into online when using source generators. 

I created a diagnostic error/report here because we need to fail at this point with some kind of message, but I'm not sure how to make it meaningful or actionable

Reply:
In that case, should we simply return? (No code is generated) instead of reporting an diagnostic error? Or we could consider reporting this as a Warning?

Reply:
Let's report it as a warning and see if this scenario ever pops up for us